### ISSUE TITLE
Action level concurrency

### ISSUE THREAD

**Issue by YadavRatnesh** [apache/openwhisk#4811]
We are using OpenWhisk for serverless service. OpenWhisk does not support action level concurrency, in order to achieve action level concurrency we are thinking of assigning each function its own namespace and set concurrency limits accordingly. 
Any issue with this approach? or any time line on action level concurrency support in OpenWhish?

**Comment by dgrove-oss** [comment]
I changed title to 'action level concurrency limits' since I think that is what you are asking about.  Actions are already executed concurrently, but the concurrency limits are managed at the namespace level, not the action level.

**Comment by michaelmarth** [comment]
@YadavRatnesh in case you use Node-based actions concurrent execution of the same action within the same container is supported, see https://github.com/apache/openwhisk/blob/master/docs/concurrency.md

**Comment by bdoyle0182** [comment]
@dgrove-oss yes that is what he's asking about. We're looking for support for more fine grained concurrency management by being able to manage concurrency limits at the action level rather than the namespace level. Without this we're contemplating reworking our data model to have one action per namespace to achieve this and then have our own external tenant level limit mangement. So we're more asking has this ever been discussed as something that might be implemented eventually or how hard would it be to implement if it's something we could potentially contribute to.
